### PR TITLE
Minor fixes to documentation outline

### PR DESCRIPTION
**Pull request by owen-mc-diffblue**
Moved a few headings around, removed "To be documented" in a few places where I don't think anything needs to be written (for non-leaf nodes in the hierarchy) and tidied up a few heading names

### CODE REVIEW

**Review by martin-cs — APPROVED** [review]

File: doc/architectural/cbmc-architecture.md [hunk 1]
```diff
@@ -59,26 +59,24 @@ transformation tools (see \ref section-other-tools).
 
 To be documented.
 
-## Instrumentation (for test gen & CBMC & ...): Goto functions -> goto functions ##
+## Instrumentation: goto functions → goto functions ##
 
 To be documented.
 
-## Goto functions -> BMC -> Counterexample (trace) ##
+## Goto functions → BMC → Counterexample (trace) ##
```
Comment (chrisr-diffblue) [inline doc/architectural/cbmc-architecture.md]: Are these arrow symbols safe across encodings/native locales? e.g. what happens on a JIS system? do the get correctly translated into HTML entities? (sorry if this is a can of worms)

**Review by chrisr-diffblue — COMMENTED** [review]
LGTM but with one query regarding the use of arrow symbols in cbmc-architecture.md

File: doc/architectural/cbmc-architecture.md (hunk 1 above)
Comment (owen-mc-diffblue) [reply doc/architectural/cbmc-architecture.md]: No idea :). It works in the HTML. I'm happy to change them all to "->" - I just want them to be consistent.

Comment (tautschnig) [reply doc/architectural/cbmc-architecture.md]: A quick web search suggests the way to (portably) do this in Markdown is `&rarr;` - could you please give it a try?

Comment (owen-mc-diffblue) [reply doc/architectural/cbmc-architecture.md]: Done

**Comment by owen-mc-diffblue** [comment]
@chrisr-diffblue I have updated with @tautschnig's preferred `&rarr;` solution

**Review by chrisr-diffblue — APPROVED** [review]

**Pull request merged**